### DISCUSSION TITLE
How to map key to toggle between executing two commands

### DISCUSSION THREAD

**Discussion by fritzrehde** [sxyazi/yazi#496] [1. Q&A]
Currently I have:
```toml
[manager]

keymap = [
	{ on = [ "s" ], exec = "sort size --reverse",                          desc = "Sort by size (reverse)" },
	{ on = [ "S" ], exec = "sort alphabetical --dir-first",                desc = "Sort alphabetically" },
]
```
but, if possible, I would like to map only `s` to toggle between the two commands. An example of the behaviour I am looking for would be: The initial state would be configured in `yazi.toml` (e.g. to the same as `S` is mapped to). Then, I press `s`, and `sort size --reverse` is executed, then I press `s` again and `sort alphabetical --dir-first` is executed, then I press `s` again and `sort size --reverse` is executed etc.
I feel like this might be possible with the new lua plugin system. Maybe you can help me with the right syntax for th plugin? Thanks!

**Comment by TobisLee** [comment]
Try this to see if it works for you:

```lua
local function entry(st)
	if st.sort_by_size then
		ya.manager_emit("sort", { "alphabetical", dir_first = true })
		st.sort_by_size = false
	else
		ya.manager_emit("sort", { "size", reverse = true })
		st.sort_by_size = true
	end
end

return {
	entry = entry,
}

```

1. Create the `resort.yazi` (you can change the dir name) in your yazi plugin dir (ie: `.config/yazi/plugins`)
2. Create the `init.lua` in `resort.yazi` and paste the above code in it.
3. Open the `keymap.toml` and paste the following code in it
```toml
[[manager.prepend_keymap]]
on   = ["s"]
run  = "plugin --sync resort"
desc = "Toggle resort"
```
